liquidation-worker-support/src/lib.rs

```diff
@@ -0,0 +1,1364 @@
+		}
+	}
+
+	/// Get the list of reserve assets.
+	/// Calls Runtime API.
+	fn fetch_reserves_list(mm_pool: EvmAddress, caller: EvmAddress) -> Result<Vec<EvmAddress>, LiquidationError> {
+		let data = Into::<u32>::into(Function::GetReservesList).to_be_bytes().to_vec();
+		let gas_limit = U256::from(500_000);
+
+		let call_info = Runtime::call(
+			caller,
+			mm_pool,
+			data,
+			U256::zero(),
+			gas_limit,
+			None,
+			None,
+			None,
+			true,
+			None,
+		)?;
+
+		if call_info.exit_reason == Succeed(Returned) {
+			let decoded = ethabi::decode(
+				&[ethabi::ParamType::Array(Box::new(ethabi::ParamType::Address))],
+				&call_info.value,
+			)
+			.map_err(LiquidationError::EthAbiError)?;
+
+			let decoded = decoded[0].clone().into_array().ok_or(ethabi::Error::InvalidData)?;
+			let mut address_arr = Vec::new();
+			for i in decoded.iter() {
+				address_arr.push(i.clone().into_address().ok_or(ethabi::Error::InvalidData)?);
+			}
+
+			Ok(address_arr)
+		} else {
+			Err(LiquidationError::EvmError(call_info.exit_reason))
+		}
+	}
+
+	/// Calls Runtime API.
+	fn fetch_reserve_data(
+		mm_pool: EvmAddress,
+		asset_address: EvmAddress,
+		caller: EvmAddress,
+	) -> Result<ReserveData, LiquidationError> {
+		let mut data = Into::<u32>::into(Function::GetReserveData).to_be_bytes().to_vec();
+		data.extend_from_slice(H256::from(asset_address).as_bytes());
+
+		let gas_limit = U256::from(500_000);
+		let call_info = Runtime::call(
+			caller,
+			mm_pool,
+			data,
+			U256::zero(),
+			gas_limit,
+			None,
+			None,
+			None,
+			true,
+			None,
+		)?;
+
+		if call_info.exit_reason == Succeed(Returned) {
+			let decoded = ethabi::decode(
+				&[
+					ethabi::ParamType::Uint(32), // ReserveConfigurationMap
+					ethabi::ParamType::Uint(16), // liquidityIndex
+					ethabi::ParamType::Uint(16), // currentLiquidityRate
+					ethabi::ParamType::Uint(16), // variableBorrowIndex
+					ethabi::ParamType::Uint(16), // currentVariableBorrowRate
+					ethabi::ParamType::Uint(16), // currentStableBorrowRate
+					ethabi::ParamType::Uint(5),  // lastUpdateTimestamp
+					ethabi::ParamType::Uint(2),  // id
+					ethabi::ParamType::Address,  // aTokenAddress
+					ethabi::ParamType::Address,  // stableDebtTokenAddress
+					ethabi::ParamType::Address,  // variableDebtTokenAddress
+					ethabi::ParamType::Address,  // interestRateStrategyAddress
+					ethabi::ParamType::Uint(16), // accruedToTreasury
+					ethabi::ParamType::Uint(16), // unbacked
+					ethabi::ParamType::Uint(16), // isolationModeTotalDebt
+				],
+				&call_info.value,
+			)?;
+
+			Ok(ReserveData::new(&decoded).ok_or(ethabi::Error::InvalidData)?)
+		} else {
+			Err(LiquidationError::EvmError(call_info.exit_reason))
+		}
+	}
+
+	/// Calls Runtime API.
+	fn fetch_asset_symbol(asset_address: &EvmAddress, caller: EvmAddress) -> Result<Vec<u8>, LiquidationError> {
+		let data = Into::<u32>::into(Function::Symbol).to_be_bytes().to_vec();
+		let gas_limit = U256::from(500_000);
```

Comment:
is this equal to TRADE_GAS_LIMIT? https://github.com/galacticcouncil/hydration-node/blob/89e1cfc425d8f5d8274b3075580ba3e5f47ee991/runtime/hydradx/src/evm/aave_trade_executor.rs#L115

If so I would use the constant somehow to make sure we don't forget to update this if the TRADE_GAS_LIMIT increases.

Reply:
We call multiple methods from different contracts, that's why there is no constant. I don't expect these limits (or gas required for execution) to change in the future. This would require deployment of new contracts.